src/App.tsx

```diff
@@ -323,24 +320,24 @@ const App = (): JSX.Element => {
     };
   }, [dispatch, bridgeLoaded, address, wrappedTokenBalance, wrappedTokenTransferableBalance]);
 
+  const queryClient = useQueryClient();
+  const governanceTokenBalanceQueryKey = useGovernanceTokenBalanceQueryKey();
+
   // Subscribes to governance token balance
   React.useEffect(() => {
-    if (!dispatch) return;
     if (!bridgeLoaded) return;
     if (!address) return;
+    if (!queryClient) return;
+    if (!governanceTokenBalanceQueryKey) return;
 
     (async () => {
       try {
         const unsubscribe = await window.bridge.tokens.subscribeToBalance(
           GOVERNANCE_TOKEN,
           address,
-          (_: string, balance: ChainBalance<GovernanceUnit>) => {
-            if (!balance.free.eq(governanceTokenBalance)) {
-              dispatch(updateGovernanceTokenBalanceAction(balance.free));
-            }
-            if (!balance.transferable.eq(governanceTokenTransferableBalance)) {
-              dispatch(updateGovernanceTokenTransferableBalanceAction(balance.transferable));
-            }
+          // TODO: it looks like the callback is called just before the balance is updated
+          () => {
+            queryClient.invalidateQueries(governanceTokenBalanceQueryKey);
```

Comment:
NIT: Im wondering if you could simply return a function from `useGovernanceTokenBalanceQueryKey` that would trigger this instead of making the hook return the params for this.

Reply:
This because the solely purpose of that hook is to provide params for this `invalidateQueries`.

Take this comments with a gain of salt btw

Reply:
@danielsimao 
Thank you for your feedback. That makes a lot of sense. I'm going to handle it once I have discussed a general direction of this PR with Tom.

Reply:
Done!